src/charm.py

```diff
@@ -39,6 +40,10 @@ def __init__(self, *args):
         if not self.public_url.startswith(("http://", "https://")):
             self.public_url = f"http://{self.public_url}"
 
+        self.public_url = self.public_url.rstrip("/")
+        if not self.public_url.endswith("/dex"):
+            self.public_url += "/dex"
```

Comment:
does that mean that we can work only with dex then?

or was it always the case?

Reply:
we've only ever tried to work with dex, but you're right this really codifies that.  Maybe there's a better way to maintain this default/backward compatibility but not be so rigid.

Practically speaking it might not be worth the effort, depending on when we plan to replace this with the identity team's tooling